src/scene/materials/material.js

```diff
@@ -159,72 +160,85 @@ class Material {
                 break;
             case BLEND_NORMAL:
+                this.separateAlphaBlend = false;
                 this.blend = true;
                 this.blendSrc = BLENDMODE_SRC_ALPHA;
                 this.blendDst = BLENDMODE_ONE_MINUS_SRC_ALPHA;
                 this.blendEquation = BLENDEQUATION_ADD;
                 break;
             case BLEND_PREMULTIPLIED:
+                this.separateAlphaBlend = false;
                 this.blend = true;
                 this.blendSrc = BLENDMODE_ONE;
                 this.blendDst = BLENDMODE_ONE_MINUS_SRC_ALPHA;
                 this.blendEquation = BLENDEQUATION_ADD;
                 break;
             case BLEND_ADDITIVE:
+                this.separateAlphaBlend = false;
                 this.blend = true;
                 this.blendSrc = BLENDMODE_ONE;
                 this.blendDst = BLENDMODE_ONE;
                 this.blendEquation = BLENDEQUATION_ADD;
                 break;
             case BLEND_ADDITIVEALPHA:
+                this.separateAlphaBlend = false;
                 this.blend = true;
                 this.blendSrc = BLENDMODE_SRC_ALPHA;
                 this.blendDst = BLENDMODE_ONE;
                 this.blendEquation = BLENDEQUATION_ADD;
                 break;
             case BLEND_MULTIPLICATIVE2X:
+                this.separateAlphaBlend = false;
                 this.blend = true;
                 this.blendSrc = BLENDMODE_DST_COLOR;
                 this.blendDst = BLENDMODE_SRC_COLOR;
                 this.blendEquation = BLENDEQUATION_ADD;
                 break;
             case BLEND_SCREEN:
+                this.separateAlphaBlend = false;
                 this.blend = true;
                 this.blendSrc = BLENDMODE_ONE_MINUS_DST_COLOR;
                 this.blendDst = BLENDMODE_ONE;
                 this.blendEquation = BLENDEQUATION_ADD;
                 break;
             case BLEND_MULTIPLICATIVE:
+                this.separateAlphaBlend = false;
                 this.blend = true;
                 this.blendSrc = BLENDMODE_DST_COLOR;
                 this.blendDst = BLENDMODE_ZERO;
                 this.blendEquation = BLENDEQUATION_ADD;
                 break;
             case BLEND_MIN:
+                this.separateAlphaBlend = false;
                 this.blend = true;
                 this.blendSrc = BLENDMODE_ONE;
                 this.blendDst = BLENDMODE_ONE;
                 this.blendEquation = BLENDEQUATION_MIN;
                 break;
             case BLEND_MAX:
+                this.separateAlphaBlend = false;
                 this.blend = true;
                 this.blendSrc = BLENDMODE_ONE;
                 this.blendDst = BLENDMODE_ONE;
                 this.blendEquation = BLENDEQUATION_MAX;
                 break;
+            case BLEND_CUSTOM:
+                Debug.warn('Use setCustomBlendType() to change blendType to BLEND_CUSTOM');
+                return;
         }
         if (prevBlend !== this.blend) {
```

Comment:
some of this code might need to execute inside setCustomBlendType I think

Reply:
adding screenshot as it's not otherwise very clear what code I'm referring to.

<img width="368" alt="Screenshot 2022-03-10 at 12 24 57" src="https://user-images.githubusercontent.com/59932779/157661301-6eeb3828-fcfa-45a8-b359-55af66bc9619.png">